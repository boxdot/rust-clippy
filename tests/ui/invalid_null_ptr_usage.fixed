// run-rustfix

fn main() {
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::NonNull::dangling().as_ptr(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::NonNull::dangling().as_ptr(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::NonNull::dangling().as_ptr(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::NonNull::dangling().as_ptr(), 0) };
}
