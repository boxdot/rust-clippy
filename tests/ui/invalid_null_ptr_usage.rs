// run-rustfix

fn main() {
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(std::ptr::null(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::null(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(std::ptr::null_mut(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::null_mut(), 0) };

    let _slice: &[usize] = unsafe { std::slice::from_raw_parts_mut(std::ptr::null_mut(), 0) };
    let _slice: &[usize] = unsafe { std::slice::from_raw_parts_mut(core::ptr::null_mut(), 0) };
}
