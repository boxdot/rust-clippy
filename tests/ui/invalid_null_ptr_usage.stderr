error: pointer must be non-null
  --> $DIR/invalid_null_ptr_usage.rs:4:64
   |
LL |     let _slice: &[usize] = unsafe { std::slice::from_raw_parts(std::ptr::null(), 0) };
   |                                                                ^^^^^^^^^^^^^^^^ help: change this to: `core::ptr::NonNull::dangling().as_ptr()`
   |
   = note: `#[deny(clippy::invalid_null_ptr_usage)]` on by default

error: pointer must be non-null
  --> $DIR/invalid_null_ptr_usage.rs:5:64
   |
LL |     let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::null(), 0) };
   |                                                                ^^^^^^^^^^^^^^^^^ help: change this to: `core::ptr::NonNull::dangling().as_ptr()`

error: pointer must be non-null
  --> $DIR/invalid_null_ptr_usage.rs:6:64
   |
LL |     let _slice: &[usize] = unsafe { std::slice::from_raw_parts(std::ptr::null_mut(), 0) };
   |                                                                ^^^^^^^^^^^^^^^^^^^^ help: change this to: `core::ptr::NonNull::dangling().as_ptr()`

error: pointer must be non-null
  --> $DIR/invalid_null_ptr_usage.rs:7:64
   |
LL |     let _slice: &[usize] = unsafe { std::slice::from_raw_parts(core::ptr::null_mut(), 0) };
   |                                                                ^^^^^^^^^^^^^^^^^^^^^ help: change this to: `core::ptr::NonNull::dangling().as_ptr()`

error: aborting due to 4 previous errors

